patches/server/0013-Paper-Plugins.patch

```diff
@@ -482,45 +483,50 @@ index 0000000000000000000000000000000000000000..85fa762d7d3476510886a7cc7b3bc79a
 +    private final Path pluginDirectory;
 +    private final Path updateDirectory;
 +
-+    PluginInitializerManager(@NotNull OptionSet minecraftOptionSet) {
++    PluginInitializerManager(final Path pluginDirectory, final Path updateDirectory) {
++        this.pluginDirectory = pluginDirectory;
++        this.updateDirectory = updateDirectory;
++    }
++
++    private static PluginInitializerManager parse(@NotNull final OptionSet minecraftOptionSet) throws Exception {
 +        // We have to load the bukkit configuration inorder to get the update folder location.
-+        File configFileLocationBukkit = (File) minecraftOptionSet.valueOf("bukkit-settings");
-+        this.pluginDirectory = ((File) minecraftOptionSet.valueOf("plugins")).toPath();
++        final File configFileLocationBukkit = (File) minecraftOptionSet.valueOf("bukkit-settings");
 +
-+        String updateDirectory = YamlConfiguration.loadConfiguration(configFileLocationBukkit).getString("settings.update-folder", "update");
-+        if (updateDirectory.isBlank()) {
-+            this.updateDirectory = null;
-+        } else {
-+            Path resolvedUpdateDirectory = this.pluginDirectory.resolve(updateDirectory);
-+            if (!Files.isDirectory(resolvedUpdateDirectory)) {
-+                this.updateDirectory = null;
-+                return;
-+            }
++        final Path pluginDirectory = ((File) minecraftOptionSet.valueOf("plugins")).toPath();
 +
-+            boolean isSameFile = true;
-+            try {
-+                isSameFile = Files.isSameFile(resolvedUpdateDirectory, this.pluginDirectory);
-+            } catch (IOException e) {
-+                LOGGER.error("Misconfigured update directory!");
-+                LOGGER.error("Failed to compare update/plugin directory", e);
-+            }
++        final YamlConfiguration configuration = PaperConfigurations.loadLegacyConfigFile(configFileLocationBukkit);
++
++        final String updateDirectoryName = configuration.getString("settings.update-folder", "update");
++        if (updateDirectoryName.isBlank()) {
++            return new PluginInitializerManager(pluginDirectory, null);
++        }
 +
-+            if (isSameFile) {
-+                LOGGER.error("Misconfigured update directory!");
-+                LOGGER.error(("Your configured update directory (%s) in bukkit.yml is pointing to the same location as the plugin directory (%s). " +
-+                    "Disabling auto updating functionality.").formatted(resolvedUpdateDirectory, this.pluginDirectory));
++        final Path resolvedUpdateDirectory = pluginDirectory.resolve(updateDirectoryName);
++        if (!Files.isDirectory(resolvedUpdateDirectory)) {
++            return new PluginInitializerManager(pluginDirectory, null);
++        }
```

Comment:
This scenario seems at least deserving of a warning log message. The update directory can be empty without a warning message, but if it points to a non directory, I think it should log a warning saying that (and saying the result will be no update directory).

Reply:
Done :+1: